Source/RunActivity/Viewer3D/Viewer.cs

```diff
@@ -547,14 +547,18 @@ public void SetCommandReceivers()
             PantographCommand.Receiver = (MSTSLocomotive)PlayerLocomotive;
             if (PlayerLocomotive is MSTSElectricLocomotive)
             {
-                CircuitBreakerClosingOrderCommand.Receiver = (MSTSElectricLocomotive)PlayerLocomotive;
-                CircuitBreakerClosingOrderButtonCommand.Receiver = (MSTSElectricLocomotive)PlayerLocomotive;
-                CircuitBreakerOpeningOrderButtonCommand.Receiver = (MSTSElectricLocomotive)PlayerLocomotive;
-                CircuitBreakerClosingAuthorizationCommand.Receiver = (MSTSElectricLocomotive)PlayerLocomotive;
+                CircuitBreakerClosingOrderCommand.Receiver = (PlayerLocomotive as MSTSLocomotive).LocomotivePowerSupply;
+                CircuitBreakerClosingOrderButtonCommand.Receiver = (PlayerLocomotive as MSTSLocomotive).LocomotivePowerSupply;
+                CircuitBreakerOpeningOrderButtonCommand.Receiver = (PlayerLocomotive as MSTSLocomotive).LocomotivePowerSupply;
+                CircuitBreakerClosingAuthorizationCommand.Receiver = (PlayerLocomotive as MSTSLocomotive).LocomotivePowerSupply;
```

Comment:
There's no benefit to the `as MSTSLocomotive` construction unless you also test for `null` when indexing into it. If the cast cannot be performed, both
```C#
(PlayerLocomotive as MSTSLocomotive).Xxx
```
and
```C#
(MSTSLocomotive)PlayerLocomotive.Xxx
```
will result in a runtime error.

Not going to change it, as this entire section needs to be refactored, really - just for future reference.

Reply:
I think that, in this case, this should be cleaner:
`if (PlayerLocomotive is MSTSElectricLocomotive electricLocomotive)` and use electricLocomotive in the code after.

About the runtime error, I don't think it can happen: we test that the locomotive is a MSTSElectricLocomotive and LocomotivePowerSupply is always initialized in the constructor of this class.